fictadvisor-api/src/v2/api/controllers/DateController.ts

```diff
@@ -15,28 +16,20 @@ export class DateController {
   ) {}
 
   @Get('/semesters')
-  @ApiOkResponse({
-    type: SemestersResponse,
-  })
-  @ApiQuery({
-    name: 'isFinished',
-    type: Boolean,
-    required: false,
+  @ApiEndpoint({
+    summary: 'Gets recent semesters',
+    documentation: DateDocumentation.SEMESTERS,
   })
   async getPreviousSemesters (
-    @Query('isFinished', ConvertToBooleanPipe) isFinished?
+    @Query('isFinished', ConvertToBooleanPipe) isFinished?,
   ) {
```

Comment:
- add `boolean` type annotation
- use built-in `ParseBoolPipe` instead of `ConvertToBooleanPipe`
- delete `ConvertToBooleanPipe`